rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -57,44 +59,79 @@ def replication_enabled(self):
         """Check whether replication enabled."""
         return ENVIRONMENT.get_value("REPLICATION_TO_RELATION_ENABLED", default=False, cast=bool)
 
-    def generate_relations_from_current_state_of_role(self):
+    def get_current_role_relations(self):
+        """Get current roles relations."""
+        return self.current_role_relations
+
+    def load_relations_from_current_state_of_role(self):
         """Generate relations from current state of role and UUIDs for v2 role and role binding from database."""
         if not self.replication_enabled():
             return
         try:
             logger.info(
                 "[Dual Write] Generate relations from current state of role(%s): '%s'", self.role.uuid, self.role.name
             )
-            relations = migrate_role(
-                self.role, False, str(self.root_workspace.uuid), self.org_id, True, True, True, False
+
+            self.binding_mapping = self.role.binding_mapping
+
+            if self.binding_mapping is None:
+                raise Exception(f"BindingMapping for role {self.role.uuid} not found.")
+
+            relations, _ = migrate_role(
+                self.role,
+                write_relationships=False,
+                root_workspace=str(self.root_workspace.uuid),
+                default_workspace=self.org_id,
+                current_bindings=self.binding_mapping,
             )
+
             self.current_role_relations = relations
         except Exception as e:
             raise DualWriteException(e)
 
-    def regenerate_relations_and_mappings_for_role(self):
-        """Delete and generated relations with mapping for a role."""
+    def generate_replication_event_to_outbox(self, role):
+        """Generate replication event to outbox table."""
         if not self.replication_enabled():
-            return []
-        return self.generate_relations_and_mappings_for_role()
```

Comment:
I inlined this since all it does really is call `generate_relations_and_mappings_for_role` and it already has the replication_enabled check, too.